### ISSUE TITLE
MPSPretrain: Config and int of loss fn and optimizer

### ISSUE THREAD

**Issue by MartinLNG** [MartinLNG/adversarial_training_mps#40]
Decide where to initialize loss function and optimizer in the MPS pretraining. It has to be configurable,
Options are: Experiment script, training loop function, training step function.
